centaur/src/it/scala/centaur/AbstractCentaurTestCaseSpec.scala

```diff
@@ -120,66 +119,67 @@ abstract class AbstractCentaurTestCaseSpec(cromwellBackends: List[String], cromw
+      case Nil => runOrDont(itShould, ignore, testCase, retries, runTest)
+      case head :: Nil => runOrDont(itShould taggedAs head, ignore, testCase, retries, runTest)
+      case head :: tail => runOrDont(itShould taggedAs(head, tail: _*), ignore, testCase, retries, runTest)
     }
   }
 
   private def runOrDont(itVerbString: ItVerbString,
                         ignore: Boolean,
-                        testName: String,
+                        testCase: CentaurTestCase,
                         retries: Int,
                         runTest: => IO[SubmitResponse]): Unit = {
     if (ignore) {
       itVerbString ignore Future.successful(succeed)
     } else {
-      itVerbString in tryTryAgain(testName, runTest, retries).unsafeToFuture().map(_ => succeed)
+      itVerbString in tryTryAgain(testCase, runTest, retries).unsafeToFuture().map(_ => succeed)
     }
   }
 
   private def runOrDont(itVerbStringTaggedAs: ItVerbStringTaggedAs,
                         ignore: Boolean,
-                        testName: String,
+                        testCase: CentaurTestCase,
                         retries: Int,
                         runTest: => IO[SubmitResponse]): Unit = {
     if (ignore) {
       itVerbStringTaggedAs ignore Future.successful(succeed)
     } else {
       itVerbStringTaggedAs in
-        tryTryAgain(testName, runTest, retries).unsafeToFuture().map(_ => succeed)
+        tryTryAgain(testCase, runTest, retries).unsafeToFuture().map(_ => succeed)
     }
   }
 
   /**
     * Returns an IO effect that will recursively try to run a test.
     *
-    * @param testName Name of the ScalaTest.
+    * @param testCase CentaurTestCase.
     * @param runTest Thunk to run the test.
     * @param retries Total number of attempts to retry.
     * @param attempt Current zero based attempt.
     * @return IO effect that will run the test, possibly retrying.
     */
-  private def tryTryAgain(testName: String, runTest: => IO[SubmitResponse], retries: Int, attempt: Int = 0): IO[SubmitResponse] = {
+  private def tryTryAgain(testCase: CentaurTestCase, runTest: => IO[SubmitResponse], retries: Int, attempt: Int = 0): IO[SubmitResponse] = {
+
     def maybeRetry(centaurTestException: CentaurTestException): IO[SubmitResponse] = {
-      val testEnvironment = TestEnvironment(testName, retries, attempt)
+
+      def clearCachedResults(workflowId: WorkflowId): IO[Unit] = CromwellDatabaseCallCaching.clearCachedResults(workflowId.toString)
+
+      val testEnvironment = TestEnvironment(testCase, retries, attempt)
       for {
         _ <- ErrorReporters.logFailure(testEnvironment, centaurTestException)
         r <- if (attempt < retries) {
-          centaurTestException
-            .workflowIdOption
-            .map(CromwellDatabaseCallCaching.clearCachedResults)
-            .getOrElse(IO.unit) *>
-          tryTryAgain(testName, runTest, retries, attempt + 1)
+          testCase.submittedWorkflowTracker.cleanUpBeforeRetry(clearCachedResults) *>
```

Comment:
TOL: Not a new problem, but I wonder what would happen if we're retrying because Centaur decided the workflow was taking too long, but then the workflow actually finishes successfully. 🤔

Reply:
I don't think a workflow running too long is a Centaur failure mode. Builds running too long are a *Travis* failure mode, but if that happens it's game over anyway...